framework/auth/views.py

```diff
@@ -36,95 +36,71 @@
 
 
 @collect_auth
-def reset_password_get(auth, verification_key=None, **kwargs):
+def reset_password_get(auth, uid=None, token=None):
     """
     View for user to land on the reset password page.
     HTTp Method: GET
 
-    :raises: HTTPError(http.BAD_REQUEST) if verification_key is invalid
+    :param auth: the authentication state
+    :param uid: the user id
+    :param token: the token in verification key
+    :return
+    :raises: HTTPError(http.BAD_REQUEST) if verification key for the user is invalid or has expired
     """
 
-    # If user is already logged in, log user out
+    # if users are logged in, log them out and redirect back to this page
     if auth.logged_in:
```

Comment:
"log user out and redirect back" is more secure and less confusing than just "go to dashboard"